data/registry_model/registry_proxy_model.py

```diff
@@ -131,6 +130,66 @@ def lookup_repository(
             state=repo.state if repo is not None else None,
         )
 
+    def _check_blob_upload_and_prune(
+        self, repo_ref: RepositoryReference, upstream_manifest: ManifestInterface
+    ) -> bool:
+        """
+        get namespace size and quota limit for the same to compare if the manifest/ blob size fits and
+        only then continue...
+        If overflow, then set lifetime_end_ms to a past date and then db.delete tag with least lifetime_end_ms
+        (order by) - this needs to loop till org frees up enough space - if not reject completely
+        """
+        curr_ns_size = namespacequota.get_namespace_size(repo_ref.namespace_name)
+
+        # get quota limit for org, if not set, skip auto pruning of images
+        quotas = namespacequota.get_namespace_quota_list(repo_ref.namespace_name)
+        if quotas:
+            # currently only one quota per namespace is supported
+            max_ns_quota = quotas[0].limit_bytes
+        else:
+            logger.info(f"tags: No quota configured")
+            return True
+
+        # check if given manifest_ref is a manifestList or a single manifest
+        if upstream_manifest.is_manifest_list:
+            logger.info(f"tags: given manifest is a list")
+            return True
+
+        upstream_blob_size = upstream_manifest.layers_compressed_size
+        logger.info(f"tags:upstream_manifest size: %s", upstream_blob_size)
+        logger.info(f"tags: curr_ns_size:  %s", curr_ns_size)
+        logger.info(f"tags: max_ns_quota:  %s", max_ns_quota)
+
+        if upstream_blob_size > max_ns_quota:
+            raise QuotaExceededException
```

Comment:
on this function you'll want to choose to either raise an exception or to return a boolean. the boolean and the exception are really about the same thing: whether the operation is allowed or not.